### DISCUSSION TITLE
Where to document which indexes Skywalking creates in ES and what those indexes do？

### DISCUSSION THREAD

**Discussion by Mushano** [apache/skywalking#10651] [Q&A] [answered]
Skywalking's steps for creating and using indexes in ES are like a black box, and it is important to understand the exact principles so that I can better deploy and manage ES.

For example, what indexes will the data collected by agent be stored in? What are the specific uses corresponding to the different indexes? Which ones are used for Tracing and which ones for Metrics?

![image](https://user-images.githubusercontent.com/38551745/230255766-3cfe6019-65b6-4930-ad70-b34d71707dc3.png)

I found a dozen pages of indexes created in the local ES index management, which makes me puzzled.

I would sincerely like to know where to find out when these indexes are created and what exactly they are used for, can I only find out through the source code?

Finally, thanks to all the contributors.

**Comment by wu-sheng** [comment, marked as answer]
Indices have been merged in recent several releases, unless you opened the old way manually.
And storage docs introduced all indices nowadays.

**Reply by Mushano** [reply]
Thanks for your answer, also I would like to ask another question, does SW support modifying the location of the saved indexes? For example, I want all the data related to segment to be stored in the index of the unique segment, instead of re-creating a new index based on different dates, so that I can predefine the index to be entered in ES in advance.
![image](https://user-images.githubusercontent.com/38551745/230305152-f8d0269d-03c4-4ac5-8ba9-482315d39cdc.png)

**Reply by wu-sheng** [reply]
If you want to change default behavior, change the source codes on your own. And it is your responsibility after that.

**Reply by Mushano** [reply]
ok，i will
